src/main/java/teammates/logic/core/StudentsLogic.java

```diff
@@ -398,35 +398,96 @@ private void verifyIsWithinSizeLimitPerEnrollment(List<StudentAttributes> studen
         }
     }
 
-    public void validateSections(List<StudentAttributes> studentList, String courseId) throws EntityDoesNotExistException, EnrollException {
+    /**
+     * Validates both sections and teams for any limit violations
+     * @param studentList
+     * @param courseId
+     * @throws EntityDoesNotExistException
+     * @throws EnrollException
+     */
+    public void validateSectionsAndTeams(List<StudentAttributes> studentList, String courseId) throws EntityDoesNotExistException, EnrollException {
 
-        List<StudentAttributes> mergedList = new ArrayList<StudentAttributes>();
-        List<StudentAttributes> studentsInCourse = getStudentsForCourse(courseId);
-        
-        for(StudentAttributes student : studentList) {
-            mergedList.add(student);
+        List<StudentAttributes> mergedList = getMergedList(studentList, courseId);
+
+        if (mergedList.size() < 2) { // no conflicts
+            return;
         }
 
-        for(StudentAttributes student : studentsInCourse) {
-            if(!isInEnrollList(student, mergedList)){
-                mergedList.add(student);
-            }
+        String errorMessage = "";
+        errorMessage += getSectionInvalidityInfo(mergedList);
+        errorMessage += getTeamInvalidityInfo(mergedList);
+
+        if (!errorMessage.isEmpty()) {
+            throw new EnrollException(errorMessage);
         }
 
-        if(mergedList.size() < 2){ // no conflicts
+    }
+    
+    /**
+     * Validates teams for any limit violations
```

Comment:
Based on the above reason, this could also be changed to `Validates teams for any team name violations`